bitcask/compaction.go

```
@@ -0,0 +1,282 @@
+			bc.index[lk.key] = &IndexEntry{
+				FileID:    currentFileID,
+				ValueSize: len(data),
+				ValuePos:  pos,
+				Timestamp: lk.entry.Timestamp,
+			}
+			currentEntryCount++
+		}
+
+		// Flush and close the last compacted file.
+		if currentFile != nil {
+			currentFile.Sync()
```

golangci-lint (2.12.2)

[error] Error return value of `currentFile.Sync` is not checked

(errcheck)